tests/Integration/Behaviour/Features/Context/Domain/Product/Combination/GenerateCombinationFeatureContext.php

```diff
@@ -85,17 +115,46 @@ public function assertCombinationName(string $combinationReference, string $comb
      * @param string $productReference
      * @param string $combinationReference
      */
-    public function assertDefaultCombination(string $productReference, string $combinationReference): void
+    public function assertDefaultCombinationForDefaultShop(string $productReference, string $combinationReference): void
     {
+        $this->assertDefaultCombination($productReference, $combinationReference);
+    }
+
+    /**
+     * @Then product :productReference default combination for shop :shopReference should be :combinationReference
+     *
+     * @param string $productReference
+     * @param string $shopReference
+     * @param string $combinationReference
+     */
+    public function assertDefaultCombinationForShop(
+        string $productReference,
+        string $shopReference,
+        string $combinationReference
+    ): void {
+        $this->assertDefaultCombination($productReference, $combinationReference, $shopReference);
+    }
+
+    /**
+     * @param string $productReference
+     * @param string $combinationReference
+     * @param string|null $shopReference
+     */
+    private function assertDefaultCombination(
+        string $productReference,
+        string $combinationReference,
+        ?string $shopReference = null
```

Comment:
```suggestion
        int $shopId
```

The parameter should be explicit here, it's in the earlier call that you handle the fallback to use `$this->getDefaultShopId()` It should be `assertDefaultCombinationForDefaultShop` which has the knowledge/responsibility to use the default ID as a fallback

And `assertDefaultCombinationForShop` is the one responsible for fetching the ID based on the reference